client/landing/stepper/declarative-flow/internals/components/redirect-to-step/index.tsx

```diff
@@ -11,12 +11,20 @@ interface Props {
 export const RedirectToStep: FC< Props > = ( { slug } ) => {
 	const { flow, lang = null } = useParams();
 	const isLoggedIn = useSelector( isUserLoggedIn );
+	const navigate = useNavigate();
 
 	const to = generatePath( '/:flow/:step/:lang?', {
 		flow: flow!,
 		step: slug,
 		lang: lang === 'en' || isLoggedIn ? null : lang,
 	} );
 
-	return <Navigate to={ `${ to }${ window.location.search }` } replace />;
+	const destination = `${ to }${ window.location.search }`;
+
+	// This needs to be a layout effect, otherwise an empty screen will be shown before the redirect.
+	useLayoutEffect( () => {
+		navigate( destination, { replace: true } );
+	}, [ destination, navigate ] );
```

Comment:
I wonder if this can be done even more directly be calling `navigate()` directly within the render function ...

Reply:
Nope that doesn't work. React Router emits a warning about having to be called within an effect. I think internally `useNavigate` is using some sort of ref or something that means it wants to be sure it's been initialised before you call `navigate`